### PR TITLE
Defect 1204

### PR DESCRIPTION
**Pull request by EllisHenderson**
Issue #1204

### CODE REVIEW

**Comment by bestbeforetoday** [comment]
Build failing because of the change made to the lerna dependency in the root `package.json`, which is not a fixed version.

File: packages/composer-common/lib/serializer/instancegenerator.js [hunk 1]
```diff
@@ -100,8 +100,7 @@ class InstanceGenerator {
         let type = field.getFullyQualifiedTypeName();
 
         if (ModelUtil.isPrimitiveType(type)) {
-            switch(type) {
-            case 'DateTime':
+            switch (type) {                case 'DateTime':
```
Comment (davidkel) [inline packages/composer-common/lib/serializer/instancegenerator.js]: slight formatting problem here

File: packages/composer-common/lib/serializer/instancegenerator.js [hunk 2]
```diff
@@ -124,16 +123,34 @@ class InstanceGenerator {
...
         classDeclaration = this.findConcreteSubclass(classDeclaration);
-
-        if (classDeclaration.isConcept()) {
-            let concept = parameters.factory.newConcept(classDeclaration.getNamespace(), classDeclaration.getName());
-            parameters.stack.push(concept);
-            return classDeclaration.accept(this, parameters);
+        let fqn = classDeclaration.getFullyQualifiedName();
+        // before doing any form of check for conept or class we want to check if this
+        // has been seen before or not
+        console.log(field);
```
Comment (davidkel) [inline packages/composer-common/lib/serializer/instancegenerator.js]: console.log here which should be removed

File: packages/composer-common/lib/serializer/instancegenerator.js [hunk 3]
```diff
@@ -124,16 +123,34 @@ class InstanceGenerator {
...
-            parameters.stack.push(resource);
-            return classDeclaration.accept(this, parameters);
+            if (field.isArray()){
+                return;
+            }
+            if (field.isOptional()){
+                return null;
+            }
+            throw new Error('Model is recursive.');
+            //todo
```
Comment (davidkel) [inline packages/composer-common/lib/serializer/instancegenerator.js]: comment still required ?

File: packages/composer-common/lib/serializer/valuegenerator.js [hunk 4]
```diff
@@ -194,7 +194,14 @@ class SampleValueGenerator extends EmptyValueGenerator {
      * @return {Array} an array
      */
     getArray(valueSupplier) {
-        return [valueSupplier()];
+        let value = valueSupplier();
+        console.log(value);
```
Comment (davidkel) [inline packages/composer-common/lib/serializer/valuegenerator.js]: console.log needs to be removed.

File: packages/composer-common/lib/serializer/valuegenerator.js [hunk 5]
```diff
@@ -194,7 +194,14 @@ class SampleValueGenerator extends EmptyValueGenerator {
...
+        if (value !== undefined) {
```
Comment (bestbeforetoday) [inline packages/composer-common/lib/serializer/valuegenerator.js]: Why are we creating valueSupplier functions that return `undefined`? And if one has been written to supply `undefined` as a value, why are we ignoring that?

**Pull request merged**